finesse/gui/temp_control.py

```diff
@@ -0,0 +1,359 @@
+        y1data.pop(0)
+        y2data.pop(0)
+
+        x = xdata[-1] + 10
+
+        # Basic RNG for testing
+        y1 = (60 * y1data[-1] + 50) % 70
+        y2 = (5 * y2data[-1] + 1) % 8
+
+        xdata.append(x)
+        y1data.append(y1)
+        y2data.append(y2)
+
+        self._bb_hot_line[0].set_xdata(xdata)
+        self._bb_hot_line[0].set_ydata(y1data)
+        self._bb_cold_line[0].set_xdata(xdata)
+        self._bb_cold_line[0].set_ydata(y2data)
+
+        self._ax1.relim()
+        self._ax2.xaxis.axes.relim()
+        self._ax1.autoscale()
+        self._ax2.xaxis.axes.autoscale()
+        self._ax2.set_ylim([self._ax2.get_ylim()[0], 2 * self._ax2.get_ylim()[1]])
+        self._canvas.draw()
+
+
+class DP9800(QGroupBox):
+    """Widgets to view the DP9800 properties."""
+
+    def __init__(self) -> None:
+        """Creates the widgets to monitor DP9800.
+
+        Returns:
+            None
+        """
+        super().__init__("DP9800")
+
+        layout = self._create_controls()
+        self.setLayout(layout)
+
+    def _create_controls(self) -> QGridLayout:
+
+        layout = QGridLayout()
+
+        layout.addWidget(QLabel("Pt 100"), 1, 0)
+        for i in range(1, 9):
+            wid = QLabel("CH_%d" % i)
+            wid.setAlignment(Qt.AlignmentFlag.AlignCenter)
+            layout.addWidget(wid, 0, i)
+
+        self._ch1 = QLineEdit()
+        self._ch1.setReadOnly(True)
+        layout.addWidget(self._ch1, 1, 1)
+
+        self._ch2 = QLineEdit()
+        self._ch2.setReadOnly(True)
+        layout.addWidget(self._ch2, 1, 2)
+
+        self._ch3 = QLineEdit()
+        self._ch3.setReadOnly(True)
+        layout.addWidget(self._ch3, 1, 3)
+
+        self._ch4 = QLineEdit()
+        self._ch4.setReadOnly(True)
+        layout.addWidget(self._ch4, 1, 4)
+
+        self._ch5 = QLineEdit()
+        self._ch5.setReadOnly(True)
+        layout.addWidget(self._ch5, 1, 5)
+
+        self._ch6 = QLineEdit()
+        self._ch6.setReadOnly(True)
+        layout.addWidget(self._ch6, 1, 6)
+
+        self._ch7 = QLineEdit()
+        self._ch7.setReadOnly(True)
+        layout.addWidget(self._ch7, 1, 7)
+
+        self._ch8 = QLineEdit()
+        self._ch8.setReadOnly(True)
+        layout.addWidget(self._ch8, 1, 8)
+
+        wid = QLabel("POLL")
+        wid.setAlignment(Qt.AlignmentFlag.AlignRight | Qt.AlignmentFlag.AlignVCenter)
+        layout.addWidget(wid, 0, 9, 2, 1)
+
+        self._poll_light = QLabel()
+        self._poll_light.setPixmap(QPixmap("./finesse/gui/images/poll_off.png"))
+        layout.addWidget(self._poll_light, 0, 10, 2, 1)
+
+        return layout
+
+
+class TC4820_HOT(QGroupBox):
+    """Widgets to view the TC4820 HOT properties."""
+
+    def __init__(self) -> None:
+        """Creates the widgets to control and monitor TC4820 HOT.
+
+        Returns:
+            None
+        """
+        super().__init__("TC4820 HOT")
+
+        layout = self._create_controls()
+        self.setLayout(layout)
```

Comment:
As you mention, I think this redundant. I'll remove the cold version and modify this one as:

```suggestion
class TC4820(QGroupBox):
    """Widgets to view the TC4820 properties."""

    def __init__(self, location: str) -> None:
        """Creates the widgets to control and monitor TC4820.

        Args:
            location: The location this controller is monitoring/controlling, typically "hot" or "cold".

        Returns:
            None
        """
        super().__init__(f"TC4820 {location.upper()}")

        self.location = location
        
        layout = self._create_controls()
        self.setLayout(layout)
```

Then, where relevant, you will use this as `TC4820("hot")` or `TC4820("cold")`.